server/server.go

```diff
@@ -0,0 +1,88 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package server
+
+import (
+	"errors"
+	"net/http"
+)
+
+// Response abstracts an HTTP response
+type Response interface {
+	Body() interface{}
+	Code() int
+	IsError() bool
+}
+
+type Result struct {
+	StatusCode   int
+	ResponseBody interface{}
+}
+
+func (r Result) Code() int {
```

Comment:
this getter business is arguably not very idiomatic, but since stream results are so different from other kinds of results, and go doesn't provide default implementations for interfaces or any other kind of abstractions; this is the best i could come up with

Reply:
Could you make the attributes `StatusCode` and `ResponseBody` unexported then?

Reply:
I can't with current implementation because they are used outside the package, eg `server.Result{StatusCode: http.StatusOK}`
Other ideas?

Reply:
What is the advantage of having the getters then? Why do you introduce `r.Code` instead of simply querying `r.StatusCode`? Same for `r.Body()`.